lib/ssh.js

```diff
@@ -1951,6 +1964,7 @@ SSH2Stream.prototype.authPKOK = function(keyAlgo, key) {
   buf.writeUInt32BE(keyLen, 5 + keyAlgoLen, true);
   key.copy(buf, 5 + keyAlgoLen + 4);
 
+  this._state.authsQueue.shift();
```

Comment:
Perhaps for method-specific responses like this one, we should verify that the current method in the queue is the right/expected type (e.g. `publickey` in this particular case).